src/db/ConnectionPool.ts

```diff
@@ -204,11 +207,12 @@ export class InstrumentedConnectionPool<C, CC extends ConnectionConfig> extends
       return connection;
     } catch (e) {
       this.acquireErrorsCounter.inc();
-      throw e;
+      throw new ExtendedError(`Could not aquire connection from pool ${this.name}-${this.readonly ? 'write' : 'read'} using user ${this.options.connectionConfig.user}`, e);
```

Comment:
I think name and readonly are part of config. They should be logged as the context by the SQLClient error handler.